src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #1
+     */
+
+    
+    // Prepare raw transaction to handle JoinSplits
+    CMutableTransaction mtx(tx_);
+    mtx.nVersion = 2;
+    crypto_sign_keypair(joinSplitPubKey_.begin(), joinSplitPrivKey_);
+    mtx.joinSplitPubKey = joinSplitPubKey_;
+    tx_ = CTransaction(mtx);
+
+    // Copy zinputs and zoutputs to more flexible containers
+    std::deque<SendManyInputJSOP> zInputsDeque;
+    for (auto o : z_inputs_) {
+        zInputsDeque.push_back(o);
+    }
+    std::deque<SendManyRecipient> zOutputsDeque;
+    for (auto o : z_outputs_) {
+        zOutputsDeque.push_back(o);
+    }
+
+    
+    /**
+     * SCENARIO #2
+     * 
+     * taddr -> taddrs
+     *       -> zaddrs
+     * 
+     * Note: Consensus rule states that coinbase utxos can only be sent to a zaddr.
+     *       Any change over and above the amount specified by the user will be sent
+     *       to the same zaddr the user is sending funds to. 
+     */
+    if (isfromtaddr_) {
+        add_taddr_outputs_to_tx();
+        
+        CAmount funds = selectedUTXOAmount;
+        CAmount fundsSpent = t_outputs_total + minersFee + z_outputs_total;
+        CAmount change = funds - fundsSpent;
+        
+        // If there is a single zaddr and there are coinbase utxos, change goes to the zaddr.
+        if (change > 0) {
+            if (isSingleZaddrOutput && selectedUTXOCoinbase) {
+                std::string address = std::get<0>(zOutputsDeque.front());   
+                SendManyRecipient smr(address, change, std::string());
+                zOutputsDeque.push_back(smr);
+
+                LogPrint("asyncrpc", "%s: change from coinbase utxo is also sent to the recipient (amount=%s)\n",
+                        getId().substr(0, 10),
+                        FormatMoney(change, false)
+                        );
+
+            } else if (!isSingleZaddrOutput && selectedUTXOCoinbase) {
+                // This should not happen and is not allowed
+                throw JSONRPCError(RPC_WALLET_ERROR, "Wallet selected Coinbase UTXOs as valid inputs when it should not have done");
+            } else {
+                // If there is a single zaddr and no coinbase utxos, just use a regular output for change.
+                add_taddr_change_output_to_tx(change);
+
+                LogPrint("asyncrpc", "%s: transparent change in transaction output (amount=%s)\n",
+                        getId().substr(0, 10),
+                        FormatMoney(change, false)
+                        );
+            }
+        }
+
+        // Create joinsplits, where each output represents a zaddr recipient.
+        Object obj;
+        while (zOutputsDeque.size() > 0) {
+            AsyncJoinSplitInfo info;
+            info.vpub_old = 0;
+            info.vpub_new = 0;
+            int n = 0;
+            while (n++<2 && zOutputsDeque.size() > 0) {
```

Comment:
zOutputsDeque is the queue of zaddr recipients to be paid.
n++<2 could be replaced with ZC_NUM_JS_OUTPUTS which I'll do now.